src/main/java/uk/ac/ebi/embl/template/TemplateEntryProcessor.java

```diff
@@ -228,6 +221,47 @@ private void replacePPOrganelleToken(TemplateVariables templateVariables) throws
         template = new StringBuilder(template.toString().replace(TemplateProcessorConstants.PP_ORGANELLE_TOKEN, ""));
     }
 
+    /**
+     * This method replaces {ORGANISM_NAME} with valid scientificName.
+     */
+    private void replaceOrganismToken(TemplateVariables templateVariables, TemplateProcessorResultSet templateProcessorResultSet) throws Exception {
+        if (!template.toString().contains(TemplateProcessorConstants.ORGANISM_TOKEN))
+            return;
+        TaxonHelper taxonHelper=new TaxonHelperImpl();
+        String scientificName="";
+        
+        for (String fieldName: templateVariables.getTokenNames()) {
+            if (StringUtils.equalsIgnoreCase(fieldName,TemplateProcessorConstants.ORGANISM_TOKEN)) {
+                String fieldValue = templateVariables.getTokenValue(fieldName);
+                
+                if (Utils.isValidTaxId(fieldValue)) {
+                    // Check if the passed value is a valid taxId.
+                    Taxon taxon = taxonHelper.getTaxonById(Long.valueOf(fieldValue));
+                    scientificName = taxon!=null ? taxon.getScientificName() : "";
+                }else {
+                    // Check if the passed value is a valid organismName.
+                    scientificName = taxonHelper.isAnyNameSubmittable(fieldValue) ? fieldValue : "";
```

Comment:
- It is fine to accept any name but we should store the scientific name in ```scientificName``` variable.